xwiki-platform-core/xwiki-platform-web/src/main/webapp/resources/uicomponents/viewers/history.js

```diff
@@ -0,0 +1,110 @@
+/*
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+require(['jquery', 'xwiki-events-bridge'], function($) {
+  var getRev = function () {
+    var rev1 = '';
+    var rev2 = '';
+
+    $("input[type='radio'][name='rev1'],input[type='radio'][name='rev2']").toArray().forEach(function (rev) {
+      if (rev.getValue()) {
+        if (rev.name == 'rev1') {
+          rev1 = rev.getValue();
+        } else {
+          rev2 = rev.getValue();
+        }
+      }
+    });
+
+    return {
+      'rev1': rev1,
+      'rev2': rev2
+    }
+  };
+
+  var getForm = function () {
+    return $('#historyform');
+  };
+
+  var submitForm = function (form) {
+    getForm().submit();
+  };
+
+  var displayCompareFunc = function(e) {
+    e.preventDefault();
+
+    var revs = getRev();
+    var newUrl = $('#history-compare-version').val()
```

Comment:
You can compute the URL in JavaScript with

    XWiki.currentDocument.getURL('action', 'queryString')

See https://github.com/xwiki/xwiki-platform/blob/master/xwiki-platform-core/xwiki-platform-web/src/main/webapp/resources/js/xwiki/xwiki.js#L1360 .

Reply:
Great that's what I was looking for, thanks!